vector/src/main/java/im/vector/app/features/onboarding/RegistrationActionHandler.kt

```diff
@@ -16,52 +16,90 @@
 
 package im.vector.app.features.onboarding
 
-import org.matrix.android.sdk.api.auth.registration.RegisterThreePid
+import im.vector.app.R
+import im.vector.app.core.resources.StringProvider
+import im.vector.app.core.utils.ensureTrailingSlash
+import im.vector.app.features.VectorFeatures
+import im.vector.app.features.VectorOverrides
+import im.vector.app.features.login.isSupported
+import im.vector.app.features.onboarding.ftueauth.MatrixOrgRegistrationStagesComparator
+import kotlinx.coroutines.flow.first
+import org.matrix.android.sdk.api.auth.AuthenticationService
+import org.matrix.android.sdk.api.auth.registration.FlowResult
 import org.matrix.android.sdk.api.auth.registration.RegistrationResult
-import org.matrix.android.sdk.api.auth.registration.RegistrationWizard
+import org.matrix.android.sdk.api.auth.registration.Stage
+import org.matrix.android.sdk.api.session.Session
 import javax.inject.Inject
 
-class RegistrationActionHandler @Inject constructor() {
+class RegistrationActionHandler @Inject constructor(
```

Comment:
the original implementation in `RegistrationActionHandler` has been extracted and renamed to `RegistrationWizardActionDelegate`